src/Tasks/Microsoft.NET.Build.Tasks/GetDependsOnNETStandard.netstandard.cs

```diff
@@ -0,0 +1,43 @@
+﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+#if NETCOREAPP1_0
+using System;
+using System.IO;
+using System.Reflection.Metadata;
+using System.Reflection.PortableExecutable;
+
+namespace Microsoft.NET.Build.Tasks
+{
+    public partial class GetDependsOnNETStandard
+    {
+        internal static bool GetFileDependsOnNETStandard(string filePath)
+        {
+            bool result = false;
+            using (var assemblyStream = new FileStream(filePath, FileMode.Open, FileAccess.Read, FileShare.Delete | FileShare.Read))
+            using (PEReader peReader = new PEReader(assemblyStream, PEStreamOptions.LeaveOpen))
+            {
+                if (peReader.HasMetadata)
+                {
+                    MetadataReader reader = peReader.GetMetadataReader();
+                    if (reader.IsAssembly)
+                    {
+                        foreach (var referenceHandle in reader.AssemblyReferences)
+                        {
+                            AssemblyReference reference = reader.GetAssemblyReference(referenceHandle);
+                            var referenceName = reader.GetString(reference.Name);
+
+                            if (result = referenceName.Equals(NetStandardAssemblyName, StringComparison.Ordinal))
```

Comment:
Nit: I think  it would clearer to return true here and false outside the loop. The local + assignment is hard to follow IMHO and unnecessary.